src/modules/scraper/adapter/messaging/DepositAcxPriceConsumer.ts

```diff
@@ -0,0 +1,39 @@
+import { OnQueueFailed, Process, Processor } from "@nestjs/bull";
+import { Logger } from "@nestjs/common";
+import { Job } from "bull";
+import { InjectRepository } from "@nestjs/typeorm";
+import { Repository } from "typeorm";
+import { DateTime } from "luxon";
+import { DepositAcxPriceQueueMessage, ScraperQueue } from ".";
+import { Deposit } from "../../model/deposit.entity";
+import { MarketPriceService } from "../../../market-price/services/service";
+
+@Processor(ScraperQueue.DepositAcxPrice)
+export class DepositAcxPriceConsumer {
+  private logger = new Logger(DepositAcxPriceConsumer.name);
+
+  constructor(
+    @InjectRepository(Deposit) private depositRepository: Repository<Deposit>,
+    private marketPriceService: MarketPriceService,
+  ) {}
+
+  @Process()
+  private async process(job: Job<DepositAcxPriceQueueMessage>) {
+    const { depositId } = job.data;
+    const deposit = await this.depositRepository.findOne({ where: { id: depositId } });
+
+    if (!deposit) return;
+    if (!deposit.depositDate) throw new Error("Invalid deposit");
+
+    const previousDate = DateTime.fromISO(deposit.depositDate.toISOString()).minus({ days: 1 }).toJSDate();
```

Comment:
What is this section doing?

Reply:
It takes the deposit date from the DB, subtracts 1 day before looking for the cached ACX price